editor/src/clj/editor/cljfx_form_view.clj

```diff
@@ -231,10 +232,15 @@
 
 ;; region string input
 
+(defn- text-formatter [props]
+  {:fx/type fx/ext-recreate-on-key-changed
+   :key (:value-converter props)
+   :desc (assoc props :fx/type fx.text-formatter/lifecycle)})
```

Comment:
The fix: re-create the whole `text-formatter` object when the `value-converter` changes.